src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -153,6 +154,15 @@ BinaryCompactObject::BinaryCompactObject(
         "of Layer 1 enveloping Object B requires excising the interior of "
         "Object B");
   }
+  using domain::BoundaryConditions::is_none;
+  if (outer_boundary_condition_ != nullptr and
+      ((not excise_interior_A_ and not excise_interior_B_) and
+       (inner_boundary_condition_ == nullptr or
+        not is_none(inner_boundary_condition_)))) {
+    PARSE_ERROR(
+        context,
+        "Inner boundary condition must be None if not excising both A and B");
```

Comment:
I would say "Inner boundary condition must be None if ExciseInteriorA and ExciseInteriorB are both false" to directly cite the input file options.  If you do so, don't forget to update the corresponding text in the `Catch::Matchers::Contains`

Reply:
I like this suggestion 👍 

I would also suggest moving this `if` block down below the `if` blocks on lines 166–180. This has the aesthetic appeal of checking for simple mistakes before checking for complicated ones... and the code appeal that below line 166–173 you no longer need the innermost "inner == nullptr" check because you know "(outer == nullptr) == (inner == nullptr)" and you're in an `if` block where "outer != nullptr", which implies "inner != nullptr".

Reply:
While I mostly agree, then I need to rework a lot of logic in the test cases for what different iteration of the for loop triggers what error, and that turns out to be quite tedious and time consuming